src/thread.c

```diff
@@ -164,16 +176,19 @@ void MTY_WaitableDestroy(MTY_Waitable **waitable)
 		return;
 
 	MTY_Waitable *ctx = *waitable;
```

Comment:
the optimizer is free to ruin the change you're trying to make here. to be guaranteed the results you seek, you should do an `exchange` here. i don't think matoya has `exchange` so a `CAS` will do. but, matoya has no `CAS` of a pointer, which is odd, because that's the first kind of CAS an atomics library usually gets.

Reply:
This would need combining with https://github.com/parsec-cloud/libmatoya/pull/131

Reply:
Integrated memory barriers